uint8 h
uint8 s
uint8 v